visualizations.py

```diff
@@ -123,13 +134,124 @@ def display_embedding_graph(model_builder):
         label='Embedding with Random Noise')
 
     # Add xticks on the middle of the group bars
-    plt.xlabel('Answer Number', fontweight='bold')
+    plt.xlabel('Embedding Dimension', fontweight='bold')
     plt.xticks([r + barWidth + 1 for r in range(len(bars1))],
                np.arange(1, len(bars1) + 1))
 
     plt.ylabel('Embedding Value', fontweight='bold')
     plt.title('Answer Embeddings', fontweight='bold')
 
-    # Create legend 
+    
+    plt.legend()
+
+    if len(argv) > 1:
+        plt.savefig(argv[1] + "/embedding.png")
+    else:
+        plt.savefig("embedding.png")
+    plt.show()
+    
+
+
+def display_perturbation_vs_epsilon(model_builder, ranker, answer_num, ref_num, epsilons):
+    '''
+    Calculates the difference between unperturbed input rank and fgsm/randomly 
+    perturbed input over a variety of epsilon values.
+
+    Args:
+        model_builder: The model in use.
+        ranker: Ranking object.
+        answer_num: The answer that is perturbed.
+        ref_num: The reference answer for the perturbed answer.
+        epsilons: Array containing the different values for epsilon.
+
+    Returns:
+        amount_of_perturbation_fgsm: Array containing amount of 

```

Comment:
returns aren't named. Can mention something like: returns a 2-tuple (r1, r2) where r1 is an array containing .... and r2 is an array containing ...